oidcdiscovery/keyhandler.go

```diff
@@ -0,0 +1,120 @@
+package oidcdiscovery
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/lestrrat-go/jwx/jwk"
+	"go.uber.org/ratelimit"
+	"golang.org/x/sync/semaphore"
+)
+
+type keyHandler struct {
+	sync.RWMutex
+	jwksURI            string
+	disableKeyID       bool
+	keySet             jwk.Set
+	fetchTimeout       time.Duration
+	keyUpdateSemaphore *semaphore.Weighted
+	keyUpdateChannel   chan keyUpdate
```

Comment:
Replace with anonymous struct: https://stackoverflow.com/questions/13670818/pair-tuple-data-type-in-go/13671024#13671024

Reply:
Skipping this, creates much more code.